hyperspy/tests/io/test_mrcz.py

```diff
@@ -18,134 +18,144 @@
+    for dtype in ['float32', 'int8', 'int16', 'uint16', 'complex64']:
+        for compressor in [None, 'zstd', 'lz4']:
+            for clevel in [1, 9]:
+                parameters.append([dtype, compressor, clevel])
+    return parameters
+
 
 class TestPythonMrcz:
-    
+
     def setup_method(self, method):
         pass
-    
-    def compareSaveLoad(self, testShape, dtype='int8', compressor=None, clevel = 1, do_async=False, **kwargs ):
+
+    def compareSaveLoad(self, testShape, dtype='int8', compressor=None,
+                        clevel=1, do_async=False, **kwargs):
         # This is the main function which reads and writes from disk.
         dtype = np.dtype(dtype)
         if dtype == 'float32' or dtype == 'float64':
-            testData = np.random.normal( size=testShape ).astype(dtype)
+            testData = np.random.normal(size=testShape).astype(dtype)
         elif dtype == 'complex64' or dtype == 'complex128':
-            testData = np.random.normal( size=testShape ).astype(dtype) + 1.0j*np.random.normal( size=testShape ).astype(dtype)
-        else: # integers
-            testData = np.random.randint( 10, size=testShape ).astype(dtype)
-            
-        testSignal = Signal2D( testData )
+            testData = np.random.normal(size=testShape).astype(
+                dtype) + 1.0j * np.random.normal(size=testShape).astype(dtype)
+        else:  # integers
+            testData = np.random.randint(10, size=testShape).astype(dtype)
+
+        testSignal = Signal2D(testData)
         # Unfortunately one cannot iterate over axes_manager in a Pythonic way
-        # for axis in testSignal.axes_manager: 
+        # for axis in testSignal.axes_manager:
         testSignal.axes_manager[0].name = 'z'
-        testSignal.axes_manager[0].scale = np.random.uniform( low=0.0, high=1.0 )
+        testSignal.axes_manager[0].scale = np.random.uniform(low=0.0, high=1.0)
         testSignal.axes_manager[0].units = '\AA'
         testSignal.axes_manager[1].name = 'x'
-        testSignal.axes_manager[1].scale = np.random.uniform( low=0.0, high=1.0 )
+        testSignal.axes_manager[1].scale = np.random.uniform(low=0.0, high=1.0)
         testSignal.axes_manager[1].units = '\AA'
         testSignal.axes_manager[2].name = 'y'
-        testSignal.axes_manager[2].scale = np.random.uniform( low=0.0, high=1.0 )
+        testSignal.axes_manager[2].scale = np.random.uniform(low=0.0, high=1.0)
         testSignal.axes_manager[2].units = '\AA'
 
         # Meta-data that goes into MRC fixed header
-        testSignal.metadata.set_item( 'Acquisition_instrument.TEM.beam_energy', 300.0 )
+        testSignal.metadata.set_item(
+            'Acquisition_instrument.TEM.beam_energy', 300.0)
         # Meta-data that goes into JSON extended header
-        testSignal.metadata.set_item('Acquisition_instrument.TEM.magnification', 25000)
+        testSignal.metadata.set_item(
+            'Acquisition_instrument.TEM.magnification', 25000)
```

Comment:
Is the date, time and time zone saved in the header? If so you can add it to the metadata.

Reply:
They are not stored in the  official 1024-byte header.  If you want to add them to the JSON extended meta-data you may.

Reply:
If there are not in the specification of the format, then let's forget it.